Index: ntp-4.2.4p0+dfsg/adjtimed/adjtimed.c
===================================================================
--- ntp-4.2.4p0+dfsg.orig/adjtimed/adjtimed.c	2004-02-25 06:57:37.000000000 +0100
+++ ntp-4.2.4p0+dfsg/adjtimed/adjtimed.c	2007-04-10 10:37:30.000000000 +0200
@@ -41,8 +41,6 @@
 #include "ntp_syslog.h"
 #include "ntp_stdlib.h"
 
-#include "adjtime.h"
-
 double atof (const char *);
 
 int InitClockRate (void);
Index: ntp-4.2.4p0+dfsg/include/ntp_md5.h
===================================================================
--- ntp-4.2.4p0+dfsg.orig/include/ntp_md5.h	2004-02-25 06:57:56.000000000 +0100
+++ ntp-4.2.4p0+dfsg/include/ntp_md5.h	2007-04-10 10:37:30.000000000 +0200
@@ -2,7 +2,12 @@
  * ntp_md5.h: deal with md5.h headers
  */
 
-#ifdef HAVE_MD5_H
+#ifdef OPENSSL
+# include <openssl/md5.h>
+#define MD5Init MD5_Init
+#define MD5Update MD5_Update
+#define MD5Final MD5_Final
+#elif defined(HAVE_MD5_H)
 # include <md5.h>
 #else
 # include "rsa_md5.h"
Index: ntp-4.2.4p0+dfsg/libntp/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/libntp/Makefile.am	2006-06-06 22:16:25.000000000 +0200
+++ ntp-4.2.4p0+dfsg/libntp/Makefile.am	2007-04-10 10:37:30.000000000 +0200
@@ -2,13 +2,13 @@
 AUTOMAKE_OPTIONS = ../util/ansi2knr
 noinst_LIBRARIES = libntp.a @MAKE_LIBNTPSIM@
 EXTRA_LIBRARIES = libntpsim.a
-libntp_a_SRCS = a_md5encrypt.c adjtime.c atoint.c atolfp.c atouint.c \
+libntp_a_SRCS = a_md5encrypt.c atoint.c atolfp.c atouint.c \
 	audio.c authkeys.c authreadkeys.c authusekey.c buftvtots.c \
 	caljulian.c caltontp.c calyearstart.c clocktime.c \
 	clocktypes.c decodenetnum.c dofptoa.c dolfptoa.c emalloc.c \
 	findconfig.c fptoa.c fptoms.c getopt.c hextoint.c \
 	hextolfp.c humandate.c icom.c inttoa.c iosignal.c \
-	lib_strbuf.c machines.c md5c.c memmove.c mfptoa.c \
+	lib_strbuf.c machines.c memmove.c mfptoa.c \
 	mfptoms.c mktime.c modetoa.c mstolfp.c msutotsf.c msyslog.c netof.c \
 	ntp_rfc2553.c numtoa.c numtohost.c octtoint.c prettydate.c \
 	ntp_random.c recvbuff.c refnumtoa.c snprintf.c socktoa.c socktohost.c \
Index: ntp-4.2.4p0+dfsg/libntp/a_md5encrypt.c
===================================================================
--- ntp-4.2.4p0+dfsg.orig/libntp/a_md5encrypt.c	2006-06-06 22:16:25.000000000 +0200
+++ ntp-4.2.4p0+dfsg/libntp/a_md5encrypt.c	2007-04-10 10:37:30.000000000 +0200
@@ -16,13 +16,7 @@
 #include "ntp_string.h"
 #include "ntp_stdlib.h"
 
-/* Disable the openssl md5 includes, because they'd clash with ours. */
-/* #define NO_MD5 */
-/* #define OPENSSL_NO_MD5 */
-#undef OPENSSL
-
 #include "ntp.h"
-#include "global.h"
 #include "ntp_md5.h"
 
 /*
Index: ntp-4.2.4p0+dfsg/ntpdate/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/ntpdate/Makefile.am	2007-02-19 12:02:14.000000000 +0100
+++ ntp-4.2.4p0+dfsg/ntpdate/Makefile.am	2007-04-10 10:37:30.000000000 +0200
@@ -8,7 +8,7 @@
 ntptimeset_SOURCES=	ntptimeset.c ntptime_config.c
 AM_CPPFLAGS=	-I$(top_srcdir)/include
 # LDADD might need RESLIB and ADJLIB
-LDADD=		version.o ../libntp/libntp.a
+LDADD=		version.o ../libntp/libntp.a @LCRYPTO@
 DISTCLEANFILES=	.version version.c stamp-v
 noinst_HEADERS=	ntpdate.h
 ETAGS_ARGS=	Makefile.am
Index: ntp-4.2.4p0+dfsg/ntpdc/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/ntpdc/Makefile.am	2006-12-28 13:03:51.000000000 +0100
+++ ntp-4.2.4p0+dfsg/ntpdc/Makefile.am	2007-04-10 10:41:55.000000000 +0200
@@ -7,7 +7,7 @@
 BUILT_SOURCES=	@MAKE_CHECK_LAYOUT@
 AM_CPPFLAGS=	-I$(top_srcdir)/include $(LIBOPTS_CFLAGS)
 # LDADD might need RESLIB and ADJLIB
-ntpdc_LDADD=	version.o @READLINE_LIBS@ $(LIBOPTS_LDADD) ../libntp/libntp.a
+ntpdc_LDADD=	version.o @READLINE_LIBS@ @LCRYPTO@ $(LIBOPTS_LDADD) ../libntp/libntp.a
 # ntpdc-layout doesn't need any additional libraries at all
 ntpdc_layout_LDADD=
 DISTCLEANFILES=	.version version.c
Index: ntp-4.2.4p0+dfsg/ntpq/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/ntpq/Makefile.am	2006-12-28 13:03:54.000000000 +0100
+++ ntp-4.2.4p0+dfsg/ntpq/Makefile.am	2007-04-10 10:47:03.000000000 +0200
@@ -4,7 +4,7 @@
 AM_CPPFLAGS=	-I$(top_srcdir)/include $(LIBOPTS_CFLAGS)
 
 # LDADD might need RESLIB and ADJLIB
-ntpq_LDADD=	version.o @READLINE_LIBS@ $(LIBOPTS_LDADD) ../libntp/libntp.a
+ntpq_LDADD=	version.o @READLINE_LIBS@ @LCRYPTO@ $(LIBOPTS_LDADD) ../libntp/libntp.a
 DISTCLEANFILES=	.version version.c
 noinst_HEADERS=	ntpq.h
 ETAGS_ARGS=	Makefile.am
Index: ntp-4.2.4p0+dfsg/include/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/include/Makefile.am	2006-12-28 13:03:02.000000000 +0100
+++ ntp-4.2.4p0+dfsg/include/Makefile.am	2007-04-10 10:37:30.000000000 +0200
@@ -6,12 +6,10 @@
 SUBDIRS = isc
 
 noinst_HEADERS = \
-	adjtime.h \
 	audio.h \
 	ascii.h \
 	audio.h \
 	binio.h \
-	global.h \
 	gps.h \
 	hopf6039.h \
 	icom.h \
@@ -55,9 +53,5 @@
 	parse.h \
 	parse_conf.h \
 	recvbuff.h \
-	rsa_md5.h \
-	timepps-SCO.h	\
-	timepps-Solaris.h	\
-	timepps-SunOS.h	\
 	trimble.h
 
Index: ntp-4.2.4p0+dfsg/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/Makefile.am	2007-01-30 12:02:36.000000000 +0100
+++ ntp-4.2.4p0+dfsg/Makefile.am	2007-04-10 11:31:22.000000000 +0200
@@ -11,7 +11,6 @@
 SUBDIRS+= 		\
 	scripts 	\
 	include 	\
-	ElectricFence	\
 	@ARLIB_DIR@ 	\
 	libntp		\
 	libparse	\
@@ -23,15 +22,12 @@
 	adjtimed	\
 	clockstuff	\
 	kernel		\
-	sntp		\
 	util		\
 	$(NULL)
 
 DIST_SUBDIRS=		\
 	scripts		\
 	include		\
-	ElectricFence	\
-	arlib		\
 	libntp		\
 	libopts		\
 	libparse	\
@@ -43,7 +39,6 @@
 	adjtimed	\
 	clockstuff	\
 	kernel		\
-	sntp		\
 	util		\
 	$(NULL)
 
Index: ntp-4.2.4p0+dfsg/configure.ac
===================================================================
--- ntp-4.2.4p0+dfsg.orig/configure.ac	2006-12-28 13:01:44.000000000 +0100
+++ ntp-4.2.4p0+dfsg/configure.ac	2007-04-10 11:31:07.000000000 +0200
@@ -2546,15 +2546,15 @@
  yes*-*-ultrix*) AC_MSG_WARN(*** But the expected answer is... no ***) ;;
 esac
 
-AC_MSG_CHECKING(for NeoClock4X receiver)
-AC_ARG_ENABLE(NEOCLOCK4X,
-    AC_HELP_STRING([--enable-NEOCLOCK4X], [+ NeoClock4X DCF77 / TDF receiver]),
-    [ntp_ok=$enableval], [ntp_ok=$ntp_eac])
-if test "$ntp_ok" = "yes"; then
-    ntp_refclock=yes
-    AC_DEFINE(CLOCK_NEOCLOCK4X, 1, [NeoClock4X])
-fi
-AC_MSG_RESULT($ntp_ok)
+dnl AC_MSG_CHECKING(for NeoClock4X receiver)
+dnl AC_ARG_ENABLE(NEOCLOCK4X,
+dnl     AC_HELP_STRING([--enable-NEOCLOCK4X], [+ NeoClock4X DCF77 / TDF receiver]),
+dnl     [ntp_ok=$enableval], [ntp_ok=$ntp_eac])
+dnl if test "$ntp_ok" = "yes"; then
+dnl     ntp_refclock=yes
+dnl     AC_DEFINE(CLOCK_NEOCLOCK4X, 1, [NeoClock4X])
+dnl fi
+dnl AC_MSG_RESULT($ntp_ok)
 
 AC_MSG_CHECKING(NMEA GPS receiver)
 AC_ARG_ENABLE(NMEA,
@@ -2950,21 +2950,21 @@
     ;;
 esac
 
-AC_MSG_CHECKING(WHARTON 400A Series clock)
-AC_ARG_ENABLE(WHARTON,
-    AC_HELP_STRING([--enable-WHARTON], [s WHARTON 400A Series clock]),
-    [ntp_ok=$enableval], [ntp_ok=$ntp_eapc])
-if test "$ntp_ok" = "yes"; then
-    ntp_libparse=yes
-    ntp_refclock=yes
-    AC_DEFINE(CLOCK_WHARTON_400A, 1, [WHARTON 400A Series protocol])
-fi
-AC_MSG_RESULT($ntp_ok)
-case "$ntp_ok$ntp_canparse" in
- yesno)
-    AC_MSG_ERROR(That's a parse clock and this system doesn't support it!)
-    ;;
-esac
+dnl AC_MSG_CHECKING(WHARTON 400A Series clock)
+dnl AC_ARG_ENABLE(WHARTON,
+dnl     AC_HELP_STRING([--enable-WHARTON], [s WHARTON 400A Series clock]),
+dnl     [ntp_ok=$enableval], [ntp_ok=$ntp_eapc])
+dnl if test "$ntp_ok" = "yes"; then
+dnl     ntp_libparse=yes
+dnl     ntp_refclock=yes
+dnl     AC_DEFINE(CLOCK_WHARTON_400A, 1, [WHARTON 400A Series protocol])
+dnl fi
+dnl AC_MSG_RESULT($ntp_ok)
+dnl case "$ntp_ok$ntp_canparse" in
+dnl  yesno)
+dnl     AC_MSG_ERROR(That's a parse clock and this system doesn't support it!)
+dnl     ;;
+dnl esac
 
 AC_MSG_CHECKING(VARITEXT clock)
 AC_ARG_ENABLE(VARITEXT,
@@ -4168,7 +4168,6 @@
 AC_DEFINE_DIR(NTP_KEYSDIR, sysconfdir, [Default location of crypto key info])
 
 AC_CONFIG_FILES(Makefile)
-AC_CONFIG_FILES(ElectricFence/Makefile)
 AC_CONFIG_FILES(adjtimed/Makefile)
 AC_CONFIG_FILES(clockstuff/Makefile)
 AC_CONFIG_FILES(include/Makefile)
@@ -4192,7 +4191,6 @@
 AC_CONFIG_FILES(scripts/ntp-wait,	[chmod +x scripts/ntp-wait])
 AC_CONFIG_FILES(scripts/ntpsweep,	[chmod +x scripts/ntpsweep])
 AC_CONFIG_FILES(scripts/ntptrace,	[chmod +x scripts/ntptrace])
-AC_CONFIG_FILES(scripts/ntpver,		[chmod +x scripts/ntpver])
 AC_CONFIG_FILES(scripts/plot_summary,	[chmod +x scripts/plot_summary])
 AC_CONFIG_FILES(scripts/summary,	[chmod +x scripts/summary])
 AC_CONFIG_FILES(util/Makefile)
Index: ntp-4.2.4p0+dfsg/util/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/util/Makefile.am	2006-12-28 13:04:25.000000000 +0100
+++ ntp-4.2.4p0+dfsg/util/Makefile.am	2007-04-10 10:37:30.000000000 +0200
@@ -1,5 +1,5 @@
 #AUTOMAKE_OPTIONS = ../ansi2knr no-dependencies
-AUTOMAKE_OPTIONS=	ansi2knr
+AUTOMAKE_OPTIONS=
 
 bin_PROGRAMS=	@MAKE_NTPTIME@ @MAKE_TICKADJ@ @MAKE_TIMETRIM@ \
 		ntp-keygen
Index: ntp-4.2.4p0+dfsg/scripts/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/scripts/Makefile.am	2007-01-30 12:02:37.000000000 +0100
+++ ntp-4.2.4p0+dfsg/scripts/Makefile.am	2007-04-10 10:40:25.000000000 +0200
@@ -1,5 +1,5 @@
 bin_SCRIPTS=	ntp-wait ntptrace
-noinst_SCRIPTS=	calc_tickadj checktime freq_adj html2man mkver ntpsweep ntpver plot_summary summary
+noinst_SCRIPTS=	calc_tickadj checktime freq_adj html2man mkver ntpsweep plot_summary summary
 EXTRA_DIST = VersionName fixautomakedepsmagic genCommitLog genver	\
 	hpadjtime.sh monitoring ntp-close ntp-groper ntp-restart	\
 	ntp-status rc1 rc2 stats
Index: ntp-4.2.4p0+dfsg/libparse/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/libparse/Makefile.am	2006-06-06 22:16:30.000000000 +0200
+++ ntp-4.2.4p0+dfsg/libparse/Makefile.am	2007-04-10 10:37:30.000000000 +0200
@@ -19,7 +19,6 @@
                      clk_rcc8000.c \
                      clk_hopf6021.c \
 	             clk_computime.c \
-		     clk_wharton.c \
 		     clk_varitext.c \
                      data_mbg.c \
 		     info_trimble.c \
@@ -42,7 +41,6 @@
 			    kclk_trimtaip$U.o \
 			    kclk_trimtsip$U.o \
 			    kclk_varitext$U.o \
-			    kclk_wharton$U.o \
 			    kbinio$U.o \
 			    kieee754io$U.o \
 			    kmfp_mul$U.o \
Index: ntp-4.2.4p0+dfsg/ntpd/Makefile.am
===================================================================
--- ntp-4.2.4p0+dfsg.orig/ntpd/Makefile.am	2006-12-28 13:03:26.000000000 +0100
+++ ntp-4.2.4p0+dfsg/ntpd/Makefile.am	2007-04-10 10:39:20.000000000 +0200
@@ -59,7 +59,7 @@
 	refclock_fg.c refclock_gpsvme.c refclock_heath.c refclock_hopfser.c \
 	refclock_hopfpci.c refclock_hpgps.c refclock_irig.c refclock_jjy.c \
 	refclock_jupiter.c refclock_leitch.c refclock_local.c \
-	refclock_mx4200.c refclock_neoclock4x.c \
+	refclock_mx4200.c \
 	refclock_nmea.c refclock_oncore.c refclock_palisade.c \
 	refclock_palisade.h refclock_parse.c \
 	refclock_pcf.c refclock_pst.c refclock_ripencc.c refclock_shm.c \
