Index: ntp-4.2.2.p4+dfsg/configure.ac
===================================================================
--- ntp-4.2.2.p4+dfsg.orig/configure.ac	2006-10-19 14:36:00.000000000 +0200
+++ ntp-4.2.2.p4+dfsg/configure.ac	2006-10-19 14:36:00.000000000 +0200
@@ -1783,14 +1783,14 @@
 
 AC_CACHE_CHECK([[availability of ntp_{adj,get}time()]], ac_cv_var_ntp_syscalls,
  [ac_cv_var_ntp_syscalls=no
- case "$ac_cv_func___adjtimex" in
-  yes)
-    ac_cv_var_ntp_syscalls=libc
-    ;;
-  *) case "$ac_cv_func_ntp_adjtime$ac_cv_func_ntp_gettime" in
-      yesyes)
-        ac_cv_var_ntp_syscalls=libc
-        ;;
+ case "$ac_cv_func_ntp_adjtime$ac_cv_func_ntp_gettime" in
+  yesyes)
+     ac_cv_var_ntp_syscalls=libc
+     ;;
+  *) case "$ac_cv_func___adjtimex" in
+      yes)
+         ac_cv_var_ntp_syscalls=adjtimex
+         ;;
       *) AC_EGREP_CPP(yes,
          [#include <sys/syscall.h>
 #if defined(SYS_ntp_gettime) && defined(SYS_ntp_adjtime)
