Index: ntp-4.2.2+dfsg/ntpd/ntp_intres.c
===================================================================
--- ntp-4.2.2+dfsg.orig/ntpd/ntp_intres.c	2006-06-06 20:16:40.000000000 +0000
+++ ntp-4.2.2+dfsg/ntpd/ntp_intres.c	2006-07-02 14:25:13.000000000 +0000
@@ -157,11 +157,11 @@
 struct ntp_res_t_pkt {		/* Tagged packet: */
 	void *tag;		/* For the caller */
 	u_int32 paddr;		/* IP to look up, or 0 */
-	char name[MAXHOSTNAMELEN]; /* Name to look up (if 1st byte is not 0) */
+        char *name;             /* Name to look up (if 1st byte is not 0) */
 };
 
 struct ntp_res_c_pkt {		/* Control packet: */
-	char name[MAXHOSTNAMELEN];
+	char *name;
 	u_int32 paddr;
 	int mode;
 	int version;
